guides/apm-release-process.md

```diff
@@ -0,0 +1,164 @@
+    verify tests work (after using `env.sh` to set up environment variables).
+  - verify that it installs in an application
+    * clone a copy of `bmacnaughton/todo` as the test application (or at a minimum
+    `rm -rf node_modules/*` in an existing todo installation.)
+    * use npm to install from the `tgz` file, like `npm install ../path-to-apm-repo/appoptics-apm@5.2.1.tgz`
+    * run `source env.sh key service-name` to set up `APPOPTICS_SERVICE_KEY`. It requires that `AO_TOKEN_STG`
+    or `AO_TOKEN_PROD` is already defined so as not to hardcode the token.
+    * run `source env.sh stg` in order to set up the environment to so that oboe will send to the staging server.
+    * `server.js` is a simple web server with many options. see the source for details.
+    * use curl or `github.com/bmacnaughton/multiload` in order to execute requests against the
+    server. `multiload` can generate consistently timed loads and offers a number of different
+    loads. see the help and source for details.
+    * verify on `appoptics.com`, either production or staging, that the events are being received
+    and are correct.
+
+3. PRE-PUBLISH STEPS
+    * create a release in `appoptics/appoptics-apm-node` and write appropriate release notes.
+        * detail any breaking changes.
+    * update https://github.com/librato/ao-kb-docs/tree/master/kb/apm_tracing/nodejs as needed. this is typically
+    for changes in the supported packages matrix.
+    * execute `git pull` or `git fetch` followed by a merge or rebase to make sure that there
+    have not been any changes to github that are not reflected in the local copy. if there have been
+    changes then, depending on the changes, go back to an appropriate place in the release process.
+
+4. PUBLISH
+    * PUBLISH FROM A SOLARWINDS ACCOUNT. This is so 1) the package is published by SolarWinds
+    and 2) so the final checks are done in a clean account without any development artifacts
+    laying around.
+    * my development machine has an appoptics account on it which i use for final testing. if you
+    don't have a separate account on your computer set one up. i just open a terminal window and
+    ssh into the appoptics account to get a clean context.
+    * when logged into that account clone the repo, `git clone --depth 1 https://github.com/appoptics/appoptics-apm-node.git`
+    * verify that `npm install` and `npm test` work correctly.
+    * create a `.tgz` file using `npm pack`, go to a development machine terminal window and verify
+    that the `.tgz` file can be installed and works correctly.
+    * log into npm with a SolarWinds account, e.g., `npm login bruce@solarwinds.cloud` and
+    publish from the root of the project using `npm publish --otp=xyzzyx`. (the one-time-password is
+    required if using token-based 2fa which should be required for logging into an account that
+    publishes). if it is not a production release be sure to add `--tag=rc` or `--tag=alpha` or similar.
```

Comment:
Is there some convention what tag to use, and how to see the npm package tags?

Reply:
noted and section expanded.